### DISCUSSION TITLE
Agent 실행시 aiohttp.clientPayloadError가 발생합니다.

### DISCUSSION THREAD

**Discussion by hyeyeon-sun** [lablup/contribution-academy-2021#67] [Q&A]
**Describe the problem**
agent에서  python -m ai.backend.agent.server --debug로 실행할 때,  aiohttp에서 ClientPayloadError가 발생합니다. 
aiohttp.client_exceptions.ClientPayloadError: Response payload is not completed

**To Reproduce**
cd /root/backend.ai-dev/agent
python -m ai.backend.agent.server --debug

**Expected behavior**
agent 실행

**Screenshots**
![image](https://user-images.githubusercontent.com/39080868/130192403-e8cd8b74-a09f-4502-ad42-e01e4d7e254b.png)

**Environment Description**
OS : WSL2 Ubuntu 20.04 LTS
Current shell : Ubuntu shell

**Comment by lizable** [comment]
@hyeyeon-sun
Could you find any clues about `aiohttp.client_excpetions.ClientPayloadError` ?
As you know, fixing the problem starts with a very simple approach, "Googling".

For now, I need more information about how this error occurred not only for the version of OS and what python version of pyenv your machine has been using but also docker configurations. Since executing docker desktop in Windows with WSL2 is not as same as using docker natively for instance Mac OS or Linux, It's hard to address what exactly triggered such an error.

So, I wonder what you found to solve this problem first, then we could make this problem narrower, which could reduce the total amount of time including communicating and actual solving time.

**Reply by hyeyeon-sun** [reply]
@lizable 

**environment info**
Python : 3.9.5v
pyenv : 2.0.4v
docker : 20.10.7v
docker-compose : 1.25.0v

current container list
![image](https://user-images.githubusercontent.com/39080868/130323698-72ce4b79-a79d-4c48-a928-07b7e0ac7c55.png)



At https://github.com/lablup/backend.ai-agent/commit/919a55ad54766b86c09bfc4979c7d35b43096dbe, I saw a pull request for code about avoiding clientPayloadError errors in **backend.ai-agent/ai/backend/agent/server.py:847**

This code is also included in version backend.ai 21.03.8v , but the error doesn't seem to be avoided properly, so I'm asking if you know anything.

➕ As a result of Googling, HTTP/2.0 does not support aiohttp, so I am trying to change the header, and to put {'Connection': 'keep-alive'} in the header to prevent the aiohttp connection from being disconnected. I'll post comments on the discussion as it progresses.

**Reply by hyeyeon-sun** [reply]
Error solved ! It seems that the docker on the desktop and the docker inside linux run at the same time and there is a conflict. I was able to run without any problem by removing the two duplicate dockers, installing the latest version of docker desktop and reinstalling backend.ai after rebooting!

<h3>Solution process</h3>
1. local(windows) : uninstall docker-desktop, and delete appdata/local/docker<br>
2. ubuntu20.04(wsl2): sudo apt purge docker-ce && sudo rm -r -f /bin/docker<br>
3. local(windows) : install docker desktop 3.6.0v && reboot<br>
4. ubuntu20.04 (wsl2): bash install-dev.sh<br>

<h3>Environment</h3>

**Previous**
 - WSL2 ubuntu 20.04v
 Python 3.9.5v
 pyenv 2.0.4v
 docker desktop 2.5.1v
 docker client 20.20.7v
 docker server 19.03.13v
 
  - WSL2 - ubuntu 20.04 connection in docker desktop settings && sudo apt-get install docker
 -> Duplicate installation

**Changed**
-  WSL2 ubuntu 20.04v
 Python 3.9.5v
 pyenv 2.0.4v
 docker desktop 3.6.0v (changed)
 docker client 20.10.8v (changed)
 docker server 20.10.8v (changed)

 - Install the latest version of docker desktop - ubuntu 20.04 connection &&
   reboot (In my case, docker desktop and ubuntu20.04 are connected.)
